### PR TITLE
DOC: Add more instructions for updating the whatsnew

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: doc/source/development/contributing_codebase.rst [hunk 1]
```diff
@@ -828,3 +835,10 @@ This will put the text *New in version 1.1.0* wherever you put the sphinx
 directive. This should also be put in the docstring when adding a new function
 or method (`example <https://github.com/pandas-dev/pandas/blob/v0.20.2/pandas/core/frame.py#L1495>`__)
 or a new keyword argument (`example <https://github.com/pandas-dev/pandas/blob/v0.20.2/pandas/core/generic.py#L568>`__).
+
+If your code is a bugfix, add your entry to the relevant bugfix section. Avoid
```
Comment (datapythonista) [inline doc/source/development/contributing_codebase.rst:839]: I find this a bit confusing. This section starts talking about the whatsnew, then the previous paragraph is about changes to an object docstring, and seems like here again we're talking about the whatsnew file again. I found this confusing when reading, seems like this is talking about a bugfix section and `Other` section in a docstring. Can we move this immediately after the other block you added?

File: doc/source/development/contributing_codebase.rst [hunk 2]
```diff
@@ -828,3 +835,10 @@ This will put the text *New in version 1.1.0* wherever you put the sphinx
...
+adding to the ``Other`` section; only in rare cases should entries go here.
```
Comment (datapythonista) [inline doc/source/development/contributing_codebase.rst:840]: Maybe it's just me, but the `here` in this sentence feels a bit confusing and unnatural. I'd refer to the `Other` section as `this section`, or more explicity, saying the `'Other' section` again.

**Review by datapythonista — COMMENTED** [review]
Looks good, couple of suggestions than IMO would make things clearer, but looks good.

File: doc/source/development/contributing_codebase.rst (hunk 1 above)
Comment (rhshadrach) [reply doc/source/development/contributing_codebase.rst:839]: Thanks - agreed.

File: doc/source/development/contributing_codebase.rst (hunk 2 above)
Comment (rhshadrach) [reply doc/source/development/contributing_codebase.rst:840]: Agreed that the use of here is not clear - what do you think of "there" instead?

**Review by datapythonista — APPROVED** [review]
Looks perfect now, thanks!

**Comment by jreback** [comment]
thanks @rhshadrach

**Pull request merged**